vegaprotocol/vega
# Connection tokens survive reboot

Close #7906 

## Test protocol

1. Start the wallet service
2. Connect to a wallet using "hostname1", keep the token
3. Connect to the same wallet using "hostname2", keep the token
4. Restart the wallet service
5. Try to list keys using the first token
    - You should be able to restore the connection, and get the keys back (with permission list aproval)
7. Try to list keys using the second token
    - You should **not** be ask to unlock the wallet, and get the keys (with permission list aproval)

## Added CLI

```
vega wallet session list
```

## Output format

<img width="1110" alt="Screenshot 2023-05-03 at 10 43 08" src="https://user-images.githubusercontent.com/4662842/235871208-4451b066-eabf-4ae1-9174-940dabd2fc34.png">
